---
title: "Fitting models"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Fitting models}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


cevomod fits the VAF spectra with mixtures of the power-law-shaped and binomial distributions, a model proposed by [Williams et al. (2018)](https://doi.org/10.1038/ng.3489). The power-law components model the neutral component of tumor evolution, as proposed by [Williams et al. (2016)](https://doi.org/10.1038/ng.3489) and by [Durrett (2013)](https://doi.org/10.1214/11-aap824). The binomial components model the clones and subclones, whose true cellular frequencies are diluted in DNA sequencing. Model components are fitted sequentially, the power-law components are fitted first, then the binomial ones are fitted to the residuals of the power-law ones.



# Power-law models

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
suppressPackageStartupMessages({
  library(cevomod)
  library(tidyverse)
})

cd <- test_data |> 
  prepare_SNVs()
cd
```



## Power-law components with the exponent equal to 2

Under the assumptions of the exponential tumor growth, constant mutation rate, and in the absence of selectively advantageous micro-clones, the exponent of the power-law component is equal 2 ([Williams et al. (2016)](https://doi.org/10.1038/ng.3489) and [Durrett (2013)](https://doi.org/10.1214/11-aap824)). The component is described by 
$$y(f) = \frac{\mu}{\beta n} \frac{1}{f^2}$$ 
where $\mu/\beta$ is the mutation rate per effective cell division, and $n$ is the number of bins in the spectrum. This model can be fitted with:

```{r}
cd <- cd |> 
  fit_powerlaw_tail_fixed(verbose = FALSE)

plot_models(cd)
```


## Power-law components with the optimized exponent

Then the assumptions of the exponential tumor growth and/or constant mutation rate are violated, or in the presence of selectively advantageous micro-clones, the power-law exponent may deviate from 2.
The component is then described by 
$$y(f) = \frac{\mu}{\beta n} \frac{1}{f^\alpha}$$
where $\mu/\beta$ is the mutation rate per effective cell division, and $n$ is the number of bins in the spectrum, and $\alpha$ is a constant $\in (0, \infty$. Sample 2 is an example sample in which the model assumptions are violated. An optimized power-law model may be used to measure the sample deviation from the model assumptions: 

```{r}
cd <- cd |> 
  fit_powerlaw_tail_optim(verbose = FALSE)

plot_models(cd)
```


```{r}
get_models(cd) |> 
  select(sample_id, model, component, alpha)
```

In Sample 2, the power-law exponent equals 4, much higher than in 3 other samples. This can indicate for example the increasing tumor mutation rate.




# Binomial components

In cevomod, the binomial components for clonal and subclonal variants are fitted to the positive part of the power-law model residuals. By default, they are fitted using the [BMix](https://github.com/caravagnalab/BMix) package ([Caravagna et al., 2020](https://www.nature.com/articles/s41588-020-0675-5)), although an alternative methods are available. In the default method, we randomly subsample the SNVs and Indels in each spectrum bin to the number given by the power-law component residual. Then, we employ the BMix to fit the VAF distribution of these variants with a mixture of 1 to 3 binomial distributions (clone plus subclones), accounting for the variant's sequencing depth. The best model is selected based on the Bayesian Information Criterium (BIC).

```{r}
cd <- fit_subclones(cd)

plot_models(cd)
```

If the subclones are fitted for the power-law component with the exponent equal to 2, the evolutionary parameters can be estimated using the equations from [Williams et al. (2018)](https://doi.org/10.1038/s41588-018-0128-6). In cevomod, we use the code implemented in the [MOBSTER package](https://github.com/caravagnalab/mobster/blob/master/R/evodynamics.R) to calculate these parameters.

Mutation rates can be obtained with `get_mutation_rates()`, and the selection coefficients with `get_selection_coefficients()` functions. Both functions can be run on the cevodata objects (with the correct models fitted), or on the model tibbles directly. This allows us to manually correct the model tibbles before the calculation of the evolutionary parameters.

For example, one can get filter Sample 2 out from the cevodata object:

```{r}
cd <- cd |> 
  filter(sample_id != "Sample 2") |> 
  fit_subclones(powerlaw_model_name = "powerlaw_fixed")

cd |> 
  get_models() |> 
  get_mutation_rates()

cd |> 
  get_models() |> 
  get_selection_coefficients()
```

## Alternative methods for fitting clones and subclones

There are 2 alternative methods for fitting clonal and subclonal components of the model.

- CliP (using `fit_subclones(method = "CliP")` or `fit_subclomes_clip()`) - uses the [CliP](https://github.com/wwylab/CliP) method published by ([Jiang et al., 2021](https://www.biorxiv.org/content/10.1101/2021.03.31.437383v1)). Running CliP requires that **[Apptainer](https://apptainer.org/) is installed** and **does not require installation of CliP and depenencies**. *cevomod* prepares the CliP input files, runs the container and reads the CliP output files back to *cevomod*. The container needs to be build *a priori* with `build_clip_container()`, which uses the image definition file [CliP.def](https://github.com/pawelqs/cevomod/inst/CliP.def). All of this can be done with a few lines of code:

```{r eval=FALSE}
set_containers_dir("~/containers/")
# get_containers_dir() to see the current containers dir
build_clip_container()
fit_subclones(cd, method = "CliP")

## OR the image can be built in the current working directory
build_clip_container()
fit_subclones(cd, method = "CliP")

## OR in any custom directory
build_clip_container("/custom/path/")
fit_subclones(cd, method = "CliP", clip_sif = "/custom/path/CliP.sif")
```

See the `fit_subclones()` help page for more details.

- mclust (using `fit_subclones(method = "mclust")` or `fit_subclomes_mclust()`) - fits the power-law component residuals with a Gaussian mixtures using the ([mclust](https://mclust-org.github.io/mclust/)) package ([Scrucca et al., 2016](https://journal.r-project.org/archive/2016/RJ-2016-021/index.html)). This is a faster but approximate method for recognition of clones and subclones.


# Bootstrapping

Since version 2.2.0 `fit_powerlaw_tail_optim()` function has a new `bootstraps` argument which can be either `FALSE` or an integer number indicating the number of bootstrap samples. Bootstrapping is performed using [tidyverse/rsample](https://rsample.tidymodels.org/) package and requires it to be installed.

**Bootstrapping significantly extends the model fitting time!**

If `bootstraps` is an integer greater than 1, the SNVs from the default assay are split by sample_id, resampled N times, new SFS spectra are computed, and N models are fitted for each sample. The resampled models are stored in the `<model_name>_bootstraps` slot (residuals for each resample are stored in the `misc` slot with the same name), and the summarized model with confidence intervals is stored in the `model_name` slot.

For demonstration purposes, we will bootstrap for 2 samples and with only 2 resamples.

```{r}
cd <- cd |> 
  filter(sample_id %in% cd$metadata$sample_id[1:2]) |> 
  fit_powerlaw_tail_optim(name = "bs_model", bootstraps = 2)

get_model_names(cd)
```

```{r}
get_models(cd, "bs_model_bootstraps")
```

```{r}
get_models(cd, "bs_model")
```

Resampled models can be plotted in the same way as non-resampled models using the `plot_models()` function.
