---
title: "Get Started"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Get Started}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# Data preparation

## `<cevodata>` S3 class

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Most of the cevomod functions use the `<cevodata>` S3 class which is designed to store both, the data and the cevomod result. The `<cevodata>` object can be easily created using the  `init_cevodata(name)` constructor and populated with data using the `add_*_data()` methods.

In this tutorial we will use the `test_data` dataset provided with cevomod, which contains 3 male samples and 1 female sample.

```{r setup}
suppressPackageStartupMessages({
  library(cevomod)
  library(tidyverse)
})

theme_set(theme_minimal())
snvs <- SNVs(test_data)
cnvs <- CNVs(test_data)
sample_data <- test_data$metadata
```

SNVs tibble should contain at least sample_id, chrom, pos and VAF columns. Any other columns are optional.

```{r}
snvs
```

We will use also some sample metadata which associates samples to patients and stores the data on patients' sex and samples' purity.

```{r}
sample_data
```

cevomod functions are pipe-oriented (in general), so we can create and populate the object through the small pipeline:

```{r}
cd <- init_cevodata(name = "Demo data") |> 
  add_SNV_data(snvs) |> 
  add_sample_data(sample_data)

cd
```

`name` can be any string that is informative for the user.


## Variant Frequency Spectra

*cevomod* fits models to the distributions of variant frequencies in the sample. Variant Allele Frequency is a basic measure of variant frequency provided by next-generation sequencing and it's equal to the fraction of reads supporting the alternate allele in all reads covering the mutation site

$$VAF = \frac{alt\_reads}{alt\_reads + ref\_reads}$$

VAF spectra can be plotted using the `plot_SFS()` function (SFS - *Site Frequency Spectrum*). Like most of the cevomod plotting functions, `plot_SFS()` returns a ggplot object which can be easily modified. Additional aesthetics can be added using the `aes()` and it can make use of the columns in the metadata tibble, which is left-joined to the plot data.

```{r}
plot_SFS(cd) +
  aes(fill = sex) +
  scale_fill_manual(values = c(male = "#DD4124", female = "#00496F")) +
  labs(title = "Variant Allele Frequency Spectra")
```

We can see that there are clear peaks of clonal mutations in all 4 samples. In 3 of them, the average frequency of the clonal variants equals 0.5, corresponding to a sample purity of 1. In Sample 2, the average frequency of clonal mutations is lower, which is in line with the lower purity of this sample:

```{r}
cd$metadata |> 
  select(sample_id, purity)
```

We can also notice small groups of mutations with VAF close to 1.0, which likely result from the looses of heterozygocity.


## Calculation of the Cancer Cell Fraction (optional)

Although VAF is often used to model cancer evolution from the bulk sequencing data, it is not an accurate measure of the true mutation frequency. VAF values are affected by sample purity and copy number alterations, as we saw in the VAF spectrum above. If we have the reliable measurements of sample purity and copy number values, we can take them into account and calculate the Cancer Cell Fraction (CCF), fraction of cancer cells in which the mutation is present. A convenient formula for calculating CCF was published by Dentro et al. in [*Principles of Reconstructing the Subclonal Architecture of Cancers* (2015)](https://doi.org/10.1101/cshperspect.a026625). To calculate CCF, cevomod requires the `purity` column in the metadata tibble and the CNVs tibble with the following columns:

- sample_id
- chrom
- start
- end
- total_cn - containing information about the total number of sequence copies,
- normal_cn - containing information about the sequence ploidy in normal cells.

```{r}
cnvs
```

We can add the CNV data to the cevodata object using the `add_CNV_data()` function.

```{r}
cd <- cd |> 
  add_CNV_data(cnvs)

cd
```

Now the CCF values can be calculated using the `calc_mutation_frequencies()` function:

```{r}
cd <- cd |> 
  calc_mutation_frequencies()
```

which will add CCF and CCF/2 columns to the SNVs tibble. cevomod will use the CCF/2 values by default starting from now.

```{r}
plot_SFS(cd) +
  aes(fill = sex) +
  scale_fill_manual(values = c(male = "#DD4124", female = "#00496F"))
```

In the new mutation frequency spectra, we can see that an average CCF/2 values of clonal peaks are now equal to 0.5 in all samples, and that no mutation has the CCF/2 value close to 1.


## Intervalization of variant frequencies

Before fitting the models, we need to binarize the mutation frequency values, which we can do using the `intervalize_mutation_frequencies()` function. cevomod will use CCF/2 values if CCF has been calculated before, and VAF if the CCF column does not is not found in the tibble. `intervalize_mutation_frequencies()` adds the `f` and `f_interval` column to the SNVs tibble. By default, the number of bins is equal to the median sequencing coverage of the variants in the sample. This allows to reduce aliasing noise in samples with low sequencing depth and to analyze samples with higher coverage with higher resolution. The desired number of bins can also be specified manually using the `bins` argument. We will also calculate the SFS spectra in this step, so the downstream functions do not have to do it on their own.

```{r}
cd <- cd |> 
  intervalize_mutation_frequencies() |>
  calc_SFS()

SNVs(cd) |> 
  glimpse()
```


Now we are ready to fit the cevomod models.

# Model fitting

Once the SNVs are prepared, the models can be fitted using the `fit_*()` functions. Full models consist of the power-law component and one or more binomial components, which are fitted sequentially.

In this example, we first fit the power-law model with an exponent of 2, and then the mixture of binomial models.

```{r}
cd <- cd |> 
  fit_powerlaw_tail_fixed() |> 
  fit_subclones()
```


```{r}
cd
```

A list of all fitted models can be obtained with:

```{r}
get_model_names(cd)
```

and the last fitted model is the active one:

```{r}
active_models(cd)
```

Models can be viewed with the `get_models()` function:

```{r}
get_models(cd, which = "powerlaw_fixed")
```

The active model is returned if the `which` argument is left empty:

```{r}
get_models(cd)
```

The tibble lists the components for all models and samples. Binomial components are called either the clones or the subclones. The component with the highest cellular frequency is called the clone, and the remaining components are called subclones. This distinction is important when estimating the subclonal evolutionary parameters: the main clones are not under positive selection in the tumor, so we only estimate the evolutionary parameters for the subclones.



# Plot models

The fitted models can be visualized with the `plot_models()` function:

```{r}
plot_models(cd) +
  aes(fill = sex) +
  scale_fill_manual(values = c(male = "#DD4124", female = "#00496F")) +
  labs(title = "cevomod models")
```

The model fitted to sample 2 is very inaccurate. The slope of the distribution is too steep to be approximated by the power-law exponent of 2. This sample should not be fitted with the model with the fixed power-law component. We will ignore this fit when estimating the evolutionary parameters.



# Evolutionary parameters

The evolutionary parameters for the subclones can be calculated using the equations provided by [Williams et al. (2018)](https://doi.org/10.1038/s41588-018-0128-6). Williams assumes an exponential tumor growth and a constant mutation rate, under which the power-law exponent equals 2, so we can use the equations with the models we fitted. In cevomod, we use the code and functions implemented in the [MOBSTER package](https://github.com/caravagnalab/mobster/blob/master/R/evodynamics.R) to calculate the parameters. We should not use the model for Sample 2, though. The mutation rate is severely underestimated, and the model fit is inaccurate. One should evaluate the fitted models carefully, before continuing with the analysis.

Mutation rates can be obtained with `get_mutation_rates()`, and the selection coefficients with `get_selection_coefficients()` functions. Both functions can be run on the cevodata objects (with the correct models fitted), or on the model tibbles directly. This allows us to manually correct the model tibbles before the calculation of the evolutionary parameters.

For example, one can get filter Sample 2 out from the cevodata object:

```{r}
cd |> 
  filter(sample_id != "Sample 2") |> 
  get_models() |> 
  get_mutation_rates()
```

or from the models tibble:


```{r}
cd |> 
  get_models() |> 
  filter(sample_id != "Sample 2") |> 
  get_selection_coefficients()
```


