<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cevomod">
<title>Fitting models • cevomod</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Fitting models">
<meta property="og:description" content="cevomod">
<meta property="og:image" content="https://pawelqs.github.io/cevomod/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">cevomod</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/get_started.html">Get Started</a>
    <a class="dropdown-item" href="../articles/cevodata_class.html">cevodata class</a>
    <a class="dropdown-item" href="../articles/fitting_models.html">Fitting models</a>
    <a class="dropdown-item" href="../articles/visualizations.html">Visualizations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pawelqs/cevomod/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Fitting models</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/pawelqs/cevomod/blob/HEAD/vignettes/fitting_models.Rmd" class="external-link"><code>vignettes/fitting_models.Rmd</code></a></small>
      <div class="d-none name"><code>fitting_models.Rmd</code></div>
    </div>

    
    
<p>cevomod fits the VAF spectra with mixtures of the power-law-shaped
and binomial distributions, a model proposed by <a href="https://doi.org/10.1038/ng.3489" class="external-link">Williams et al. (2018)</a>. The
power-law components model the neutral component of tumor evolution, as
proposed by <a href="https://doi.org/10.1038/ng.3489" class="external-link">Williams et
al. (2016)</a> and by <a href="https://doi.org/10.1214/11-aap824" class="external-link">Durrett (2013)</a>. The
binomial components model the clones and subclones, whose true cellular
frequencies are diluted in DNA sequencing. Model components are fitted
sequentially, the power-law components are fitted first, then the
binomial ones are fitted to the residuals of the power-law ones.</p>
<div class="section level2">
<h2 id="power-law-models">Power-law models<a class="anchor" aria-label="anchor" href="#power-law-models"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pawelqs.github.io/cevomod/">cevomod</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">test_data</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/prepare_SNVs.html">prepare_SNVs</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">0 variants (0 %), have NA CCF value</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating f intervals, using CCF/2 column</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating SFS statistics</span></span></span>
<span><span class="va">cd</span></span>
<span><span class="co">#&gt; <span style="color: #5F8700;">&lt;cevodata&gt; dataset: test_data</span></span></span>
<span><span class="co">#&gt; <span style="color: #87AF5F;">Genome: unknown</span></span></span>
<span><span class="co">#&gt; SNV assays: snvs (default)</span></span>
<span><span class="co">#&gt; CNV assays: cnvs (default)</span></span>
<span><span class="co">#&gt; 4 cases, 4 samples, 1 sample per case</span></span>
<span><span class="co">#&gt; 16000 mutations total, 4000 +/- 0 mutations per case</span></span>
<span><span class="co">#&gt; Active models:</span></span></code></pre></div>
<div class="section level3">
<h3 id="power-law-components-with-the-exponent-equal-to-2">Power-law components with the exponent equal to 2<a class="anchor" aria-label="anchor" href="#power-law-components-with-the-exponent-equal-to-2"></a>
</h3>
<p>Under the assumptions of the exponential tumor growth, constant
mutation rate, and in the absence of selectively advantageous
micro-clones, the exponent of the power-law component is equal 2 (<a href="https://doi.org/10.1038/ng.3489" class="external-link">Williams et al. (2016)</a> and <a href="https://doi.org/10.1214/11-aap824" class="external-link">Durrett (2013)</a>). The
component is described by <span class="math display">\[y(f) =
\frac{\mu}{\beta n} \frac{1}{f^2}\]</span> where <span class="math inline">\(\mu/\beta\)</span> is the mutation rate per
effective cell division, and <span class="math inline">\(n\)</span> is
the number of bins in the spectrum. This model can be fitted with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/powerlaw_fixed_model.html">fit_powerlaw_tail_fixed</a></span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Mf_1f's not calculated yet. Calculating with default bins</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating Williams's M(f) ~ 1/f statistics, using CCF/2 column</span></span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_models.html">plot_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in geom_bar(join_aes(bar_mapping, mapping), stat = "identity", alpha =</span></span>
<span><span class="co">#&gt; alpha, : Ignoring unknown aesthetics: <span style="color: #00BB00;">width</span></span></span></code></pre></div>
<p><img src="fitting_models_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="power-law-components-with-the-optimized-exponent">Power-law components with the optimized exponent<a class="anchor" aria-label="anchor" href="#power-law-components-with-the-optimized-exponent"></a>
</h3>
<p>Then the assumptions of the exponential tumor growth and/or constant
mutation rate are violated, or in the presence of selectively
advantageous micro-clones, the power-law exponent may deviate from 2.
The component is then described by <span class="math display">\[y(f) =
\frac{\mu}{\beta n} \frac{1}{f^\alpha}\]</span> where <span class="math inline">\(\mu/\beta\)</span> is the mutation rate per
effective cell division, and <span class="math inline">\(n\)</span> is
the number of bins in the spectrum, and <span class="math inline">\(\alpha\)</span> is a constant <span class="math inline">\(\in (0, \infty\)</span>. Sample 2 is an example
sample in which the model assumptions are violated. An optimized
power-law model may be used to measure the sample deviation from the
model assumptions:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/powerlaw_optim.html">fit_powerlaw_tail_optim</a></span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_models.html">plot_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in geom_bar(join_aes(bar_mapping, mapping), stat = "identity", alpha =</span></span>
<span><span class="co">#&gt; alpha, : Ignoring unknown aesthetics: <span style="color: #00BB00;">width</span></span></span></code></pre></div>
<p><img src="fitting_models_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">model</span>, <span class="va">component</span>, <span class="va">alpha</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 4</span></span></span>
<span><span class="co">#&gt;   sample_id model          component     alpha</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1  powerlaw_optim powerlaw tail  1.88</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 2  powerlaw_optim powerlaw tail  3.27</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 3  powerlaw_optim powerlaw tail  2.02</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 4  powerlaw_optim powerlaw tail  1.86</span></span></code></pre></div>
<p>In Sample 2, the power-law exponent equals 4, much higher than in 3
other samples. This can indicate for example the increasing tumor
mutation rate.</p>
</div>
</div>
<div class="section level2">
<h2 id="binomial-components">Binomial components<a class="anchor" aria-label="anchor" href="#binomial-components"></a>
</h2>
<p>In cevomod, the binomial components for clonal and subclonal variants
are fitted to the positive part of the power-law model residuals. By
default, they are fitted using the <a href="https://github.com/caravagnalab/BMix" class="external-link">BMix</a> package, although
an alternative method using mclust is also available. In the default
method, we randomly subsample the SNVs and Indels in each spectrum bin
to the number given by the power-law component residual. Then, we employ
the BMix to fit the VAF distribution of these variants with a mixture of
1 to 3 binomial distributions (clone plus subclones), accounting for the
variant’s sequencing depth. The best model is selected based on the
Bayesian Information Criterium (BIC).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_subclones.html">fit_subclones_bmix</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Fitting binomial models using BMix</span></span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span></span>
<span><span class="co">#&gt; 'BMix'</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">BMix</span>, <span style="font-style: italic;">'Binomial and Beta-Binomial univariate mixtures'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagnalab.github.io/BMix/&gt;</span></span></span>
<span><span class="co">#&gt; Warning: There was 1 warning in `reframe()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> In argument: `fit_binomial_models_BMix(.data$data, N, pb, verbose)`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> In row 1.</span></span>
<span><span class="co">#&gt; Caused by warning:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> replacing previous import 'cli::num_ansi_colors' by 'crayon::num_ansi_colors' when loading 'easypar'</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_models.html">plot_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in geom_bar(join_aes(bar_mapping, mapping), stat = "identity", alpha =</span></span>
<span><span class="co">#&gt; alpha, : Ignoring unknown aesthetics: <span style="color: #00BB00;">width</span></span></span></code></pre></div>
<p><img src="fitting_models_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>If the subclones are fitted for the power-law component with the
exponent equal to 2, the evolutionary parameters can be estimated using
the equations from <a href="https://doi.org/10.1038/s41588-018-0128-6" class="external-link">Williams et
al. (2018)</a>. In cevomod, we use the code implemented in the <a href="https://github.com/caravagnalab/mobster/blob/master/R/evodynamics.R" class="external-link">MOBSTER
package</a> to calculate these parameters.</p>
<p>Mutation rates can be obtained with
<code><a href="../reference/evo_params.html">get_mutation_rates()</a></code>, and the selection coefficients with
<code><a href="../reference/evo_params.html">get_selection_coefficients()</a></code> functions. Both functions can
be run on the cevodata objects (with the correct models fitted), or on
the model tibbles directly. This allows us to manually correct the model
tibbles before the calculation of the evolutionary parameters.</p>
<p>For example, one can get filter Sample 2 out from the cevodata
object:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample_id</span> <span class="op">!=</span> <span class="st">"Sample 2"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/fit_subclones.html">fit_subclones_bmix</a></span><span class="op">(</span>powerlaw_model_name <span class="op">=</span> <span class="st">"powerlaw_fixed"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Fitting binomial models using BMix</span></span></span>
<span></span>
<span><span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/evo_params.html">get_mutation_rates</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">#&gt;   sample_id mutation_rate_williams</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1                    46.6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 3                   104. </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 4                    78.8</span></span>
<span></span>
<span><span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/evo_params.html">get_selection_coefficients</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 8</span></span></span>
<span><span class="co">#&gt;   sample_id mutation_rate_williams component  N_mutations subclone_frequency</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 3                   104.  Subclone 1         726              0.982</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 3                   104.  Subclone 2         784              0.235</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 4                    78.8 Subclone 1         822              0.999</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 4                    78.8 Subclone 2         366              0.177</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: emergence_time &lt;dbl&gt;, time_end &lt;dbl&gt;, selection &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Paweł Kuś.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
