<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cevomod">
<title>Get Started • cevomod</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Started">
<meta property="og:description" content="cevomod">
<meta property="og:image" content="https://pawelqs.github.io/cevomod/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">cevomod</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.3.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/get_started.html">Get Started</a>
    <a class="dropdown-item" href="../articles/cevodata_class.html">cevodata class</a>
    <a class="dropdown-item" href="../articles/fitting_models.html">Fitting models</a>
    <a class="dropdown-item" href="../articles/visualizations.html">Visualizations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pawelqs/cevomod/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get Started</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/pawelqs/cevomod/blob/HEAD/vignettes/get_started.Rmd" class="external-link"><code>vignettes/get_started.Rmd</code></a></small>
      <div class="d-none name"><code>get_started.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="data-preparation">Data preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h2>
<div class="section level3">
<h3 id="cevodata-s3-class">
<code>&lt;cevodata&gt;</code> S3 class<a class="anchor" aria-label="anchor" href="#cevodata-s3-class"></a>
</h3>
<p>Most of the cevomod functions use the <code>&lt;cevodata&gt;</code>
S3 class which is designed to store both, the data and the cevomod
result. The <code>&lt;cevodata&gt;</code> object can be easily created
using the <code>init_cevodata(name)</code> constructor and populated
with data using the <code>add_*_data()</code> methods.</p>
<p>In this tutorial we will use the <code>test_data</code> dataset
provided with cevomod, which contains 3 male samples and 1 female
sample.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pawelqs.github.io/cevomod/">cevomod</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">snvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assays.html">SNVs</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span></span>
<span><span class="va">cnvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assays.html">CNVs</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span></span>
<span><span class="va">sample_data</span> <span class="op">&lt;-</span> <span class="va">test_data</span><span class="op">$</span><span class="va">metadata</span></span></code></pre></div>
<p>SNVs tibble should contain at least sample_id, chrom, pos and VAF
columns. Any other columns are optional.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snvs</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">&lt;cevo_snvs&gt; tibble</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 16,000 × 11</span></span></span>
<span><span class="co">#&gt;    sample_id chrom   pos gene_symbol ref   alt   ref_reads alt_reads impact</span></span>
<span><span class="co">#&gt;  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Sample 1  chr1      1 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           25        29 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Sample 1  chr1      2 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           22        19 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Sample 1  chr1      3 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           21        20 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Sample 1  chr1      4 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           34         2 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Sample 1  chr1      5 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           62         1 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Sample 1  chr1      6 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>            4         3 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Sample 1  chr1      7 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           33        34 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Sample 1  chr1      8 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           31        30 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Sample 1  chr1      9 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           25        17 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Sample 1  chr1     10 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           41        31 <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 15,990 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: VAF &lt;dbl&gt;, DP &lt;int&gt;</span></span></span></code></pre></div>
<p>We will use also some sample metadata which associates samples to
patients and stores the data on patients’ sex and samples’ purity.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_data</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 4</span></span></span>
<span><span class="co">#&gt;   sample_id patient_id sex    purity</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1  Patient 1  male      1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 2  Patient 2  male      0.7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 3  Patient 3  female    1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 4  Patient 4  male      1</span></span></code></pre></div>
<p>cevomod functions are pipe-oriented (in general), so we can create
and populate the object through the small pipeline:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/init_cevodata.html">init_cevodata</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Demo data"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/assays.html">add_SNV_data</a></span><span class="op">(</span><span class="va">snvs</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/cevo_metadata.html">add_sample_data</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cd</span></span>
<span><span class="co">#&gt; <span style="color: #5F8700;">&lt;cevodata&gt; dataset: Demo data</span></span></span>
<span><span class="co">#&gt; <span style="color: #87AF5F;">Genome: unknown</span></span></span>
<span><span class="co">#&gt; SNV assays: snvs (default)</span></span>
<span><span class="co">#&gt; CNV assays: None</span></span>
<span><span class="co">#&gt; 4 cases, 4 samples, 1 sample per case</span></span>
<span><span class="co">#&gt; 16000 mutations total, 4000 +/- 0 mutations per case</span></span>
<span><span class="co">#&gt; Active models:</span></span></code></pre></div>
<p><code>name</code> can be any string that is informative for the
user.</p>
<p><em>To facilitate the use of cevomod with the data from popular
variant callers such as Mutect2, Strelka2, ASCAT, or FACETS, we have
implemented a </em> <a href="https://pawelqs.github.io/readthis/index.html" class="external-link">readthis</a>
<em>package. readthis functions are designed for bulk reading of many
output variant files (they accept a path to a single file, named vector
of file paths, or a path to a directory containing many files). Data
objects read with readthis functions can be added to the cevodata object
with a single call of general</em> <code><a href="../reference/readthis-integration.html">add_data()</a></code> <em>function.
For more information see the</em> <a href="https://pawelqs.github.io/readthis/index.html" class="external-link">readthis
page</a>.</p>
</div>
<div class="section level3">
<h3 id="variant-frequency-spectra">Variant Frequency Spectra<a class="anchor" aria-label="anchor" href="#variant-frequency-spectra"></a>
</h3>
<p><em>cevomod</em> fits models to the distributions of variant
frequencies in the sample. Variant Allele Frequency is a basic measure
of variant frequency provided by next-generation sequencing and it’s
equal to the fraction of reads supporting the alternate allele in all
reads covering the mutation site</p>
<p><span class="math display">\[VAF = \frac{alt\_reads}{alt\_reads +
ref\_reads}\]</span></p>
<p>VAF spectra can be plotted using the <code><a href="../reference/sfs.html">plot_SFS()</a></code> function
(SFS - <em>Site Frequency Spectrum</em>). Like most of the cevomod
plotting functions, <code><a href="../reference/sfs.html">plot_SFS()</a></code> returns a ggplot object
which can be easily modified. Additional aesthetics can be added using
the <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code> and it can make use of the columns in the
metadata tibble, which is left-joined to the plot data.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sfs.html">plot_SFS</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>male <span class="op">=</span> <span class="st">"#DD4124"</span>, female <span class="op">=</span> <span class="st">"#00496F"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Variant Allele Frequency Spectra"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating SFS statistics</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating f intervals, using VAF column</span></span></span>
<span><span class="co">#&gt; Warning in geom_bar(join_aes(bar_mapping, mapping), stat = "identity", alpha =</span></span>
<span><span class="co">#&gt; alpha, : Ignoring unknown aesthetics: <span style="color: #00BB00;">width</span></span></span></code></pre></div>
<p><img src="get_started_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>We can see that there are clear peaks of clonal mutations in all 4
samples. In 3 of them, the average frequency of the clonal variants
equals 0.5, corresponding to a sample purity of 1. In Sample 2, the
average frequency of clonal mutations is lower, which is in line with
the lower purity of this sample:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span><span class="op">$</span><span class="va">metadata</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">purity</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 2</span></span></span>
<span><span class="co">#&gt;   sample_id purity</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1     1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 2     0.7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 3     1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 4     1</span></span></code></pre></div>
<p>We can also notice small groups of mutations with VAF close to 1.0,
which likely result from the looses of heterozygocity.</p>
</div>
<div class="section level3">
<h3 id="calculation-of-the-cancer-cell-fraction-optional">Calculation of the Cancer Cell Fraction (optional)<a class="anchor" aria-label="anchor" href="#calculation-of-the-cancer-cell-fraction-optional"></a>
</h3>
<p>Although VAF is often used to model cancer evolution from the bulk
sequencing data, it is not an accurate measure of the true mutation
frequency. VAF values are affected by sample purity and copy number
alterations, as we saw in the VAF spectrum above. If we have the
reliable measurements of sample purity and copy number values, we can
take them into account and calculate the Cancer Cell Fraction (CCF),
fraction of cancer cells in which the mutation is present. A convenient
formula for calculating CCF was published by Dentro et al. in <a href="https://doi.org/10.1101/cshperspect.a026625" class="external-link"><em>Principles of
Reconstructing the Subclonal Architecture of Cancers</em> (2015)</a>. To
calculate CCF, cevomod requires the <code>purity</code> column in the
metadata tibble and the CNVs tibble with the following columns:</p>
<ul>
<li>sample_id</li>
<li>chrom</li>
<li>start</li>
<li>end</li>
<li>total_cn - containing information about the total number of sequence
copies,</li>
<li>normal_cn - containing information about the sequence ploidy in
normal cells.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cnvs</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 8 × 8</span></span></span>
<span><span class="co">#&gt;   sample_id chrom start   end total_cn major_cn minor_cn normal_cn</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1  chr1      1  <span style="text-decoration: underline;">4</span>000        2        1        0         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 1  chr2      1  <span style="text-decoration: underline;">4</span>000        1        1        0         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 2  chr1      1  <span style="text-decoration: underline;">4</span>000        2        1        0         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 2  chr2      1  <span style="text-decoration: underline;">4</span>000        1        1        0         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Sample 3  chr1      1  <span style="text-decoration: underline;">4</span>000        2        1        0         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Sample 3  chr2      1  <span style="text-decoration: underline;">4</span>000        1        1        0         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> Sample 4  chr1      1  <span style="text-decoration: underline;">4</span>000        2        1        0         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">8</span> Sample 4  chr2      1  <span style="text-decoration: underline;">4</span>000        1        1        0         2</span></span></code></pre></div>
<p>We can add the CNV data to the cevodata object using the
<code><a href="../reference/assays.html">add_CNV_data()</a></code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/assays.html">add_CNV_data</a></span><span class="op">(</span><span class="va">cnvs</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cd</span></span>
<span><span class="co">#&gt; <span style="color: #5F8700;">&lt;cevodata&gt; dataset: Demo data</span></span></span>
<span><span class="co">#&gt; <span style="color: #87AF5F;">Genome: unknown</span></span></span>
<span><span class="co">#&gt; SNV assays: snvs (default)</span></span>
<span><span class="co">#&gt; CNV assays: cnvs (default)</span></span>
<span><span class="co">#&gt; 4 cases, 4 samples, 1 sample per case</span></span>
<span><span class="co">#&gt; 16000 mutations total, 4000 +/- 0 mutations per case</span></span>
<span><span class="co">#&gt; Active models:</span></span></code></pre></div>
<p>Now the CCF values can be calculated using the
<code><a href="../reference/calc_mutation_frequencies.html">calc_mutation_frequencies()</a></code> function:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/calc_mutation_frequencies.html">calc_mutation_frequencies</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">0 variants (0 %), have NA CCF value</span></span></span></code></pre></div>
<p>which will add CCF and CCF/2 columns to the SNVs tibble. cevomod will
use the CCF/2 values by default starting from now.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sfs.html">plot_SFS</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>male <span class="op">=</span> <span class="st">"#DD4124"</span>, female <span class="op">=</span> <span class="st">"#00496F"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating SFS statistics</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating f intervals, using CCF/2 column</span></span></span>
<span><span class="co">#&gt; Warning in geom_bar(join_aes(bar_mapping, mapping), stat = "identity", alpha =</span></span>
<span><span class="co">#&gt; alpha, : Ignoring unknown aesthetics: <span style="color: #00BB00;">width</span></span></span></code></pre></div>
<p><img src="get_started_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>In the new mutation frequency spectra, we can see that an average
CCF/2 values of clonal peaks are now equal to 0.5 in all samples, and
that no mutation has the CCF/2 value close to 1.</p>
</div>
<div class="section level3">
<h3 id="intervalization-of-variant-frequencies">Intervalization of variant frequencies<a class="anchor" aria-label="anchor" href="#intervalization-of-variant-frequencies"></a>
</h3>
<p>Before fitting the models, we need to binarize the mutation frequency
values, which we can do using the
<code><a href="../reference/intervalize_mutation_frequencies.html">intervalize_mutation_frequencies()</a></code> function. cevomod will
use CCF/2 values if CCF has been calculated before, and VAF if the CCF
column does not is not found in the tibble.
<code><a href="../reference/intervalize_mutation_frequencies.html">intervalize_mutation_frequencies()</a></code> adds the <code>f</code>
and <code>f_interval</code> column to the SNVs tibble. By default, the
number of bins is equal to the median sequencing coverage of the
variants in the sample. This allows to reduce aliasing noise in samples
with low sequencing depth and to analyze samples with higher coverage
with higher resolution. The desired number of bins can also be specified
manually using the <code>bins</code> argument. We will also calculate
the SFS spectra in this step, so the downstream functions do not have to
do it on their own.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/intervalize_mutation_frequencies.html">intervalize_mutation_frequencies</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/sfs.html">calc_SFS</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating f intervals, using CCF/2 column</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating SFS statistics</span></span></span>
<span></span>
<span><span class="fu"><a href="../reference/assays.html">SNVs</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 16,000</span></span>
<span><span class="co">#&gt; Columns: 15</span></span>
<span><span class="co">#&gt; $ sample_id   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Sample 1", "Sample 1", "Sample 1", "Sample 1", "Sample 1"…</span></span>
<span><span class="co">#&gt; $ chrom       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "chr1", "chr1", "chr1", "chr1", "chr1", "chr1", "chr1", "c…</span></span>
<span><span class="co">#&gt; $ pos         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17,…</span></span>
<span><span class="co">#&gt; $ gene_symbol <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ ref         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ alt         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ ref_reads   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 25, 22, 21, 34, 62, 4, 33, 31, 25, 41, 42, 64, 32, 29, 42,…</span></span>
<span><span class="co">#&gt; $ alt_reads   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 29, 19, 20, 2, 1, 3, 34, 30, 17, 31, 1, 1, 25, 31, 40, 1, …</span></span>
<span><span class="co">#&gt; $ impact      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ VAF         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.536, 0.462, 0.488, 0.078, 0.030, 0.444, 0.507, 0.483, 0.…</span></span>
<span><span class="co">#&gt; $ CCF         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1.072, 0.924, 0.976, 0.156, 0.060, 0.888, 1.014, 0.966, 0.…</span></span>
<span><span class="co">#&gt; $ `CCF/2`     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.536, 0.462, 0.488, 0.078, 0.030, 0.444, 0.507, 0.483, 0.…</span></span>
<span><span class="co">#&gt; $ f_interval  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "(0.529,0.549]", "(0.451,0.471]", "(0.471,0.49]", "(0.0588…</span></span>
<span><span class="co">#&gt; $ f           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.5390, 0.4610, 0.4805, 0.0686, 0.0294, 0.4410, 0.5000, 0.…</span></span>
<span><span class="co">#&gt; $ DP          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 54, 41, 41, 36, 63, 7, 67, 61, 42, 72, 43, 65, 57, 60, 82,…</span></span></code></pre></div>
<p>Now we are ready to fit the cevomod models.</p>
</div>
</div>
<div class="section level2">
<h2 id="model-fitting">Model fitting<a class="anchor" aria-label="anchor" href="#model-fitting"></a>
</h2>
<p>Once the SNVs are prepared, the models can be fitted using the
<code>fit_*()</code> functions. Full models consist of the power-law
component and one or more binomial components, which are fitted
sequentially.</p>
<p>In this example, we first fit the power-law model with an exponent of
2, and then the mixture of binomial models. By default, clonal and
subclonal components are fitted using <a href="https://github.com/caravagnalab/BMix" class="external-link">BMix</a> package (<a href="https://www.nature.com/articles/s41588-020-0675-5" class="external-link">Caravagna et
al., 2020</a>). See other methods for fitting these components in <a href="https://pawelqs.github.io/cevomod/articles/fitting_models.html#binomial-components">Fitting
models/Binomial components</a> vignette.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/powerlaw_fixed_model.html">fit_powerlaw_tail_fixed</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/fit_subclones.html">fit_subclones</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Fitting binomial models using BMix</span></span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span></span>
<span><span class="co">#&gt; 'BMix'</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">BMix</span>, <span style="font-style: italic;">'Binomial and Beta-Binomial univariate mixtures'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagnalab.github.io/BMix/&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Fitting williams neutral models...</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Mf_1f's not calculated yet. Calculating with default bins</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating Williams's M(f) ~ 1/f statistics, using CCF/2 column</span></span></span>
<span><span class="co">#&gt; Warning: There was 1 warning in `reframe()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> In argument: `fit_binomial_models_BMix(.data$data, N, pb, verbose)`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> In row 1.</span></span>
<span><span class="co">#&gt; Caused by warning:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> replacing previous import 'cli::num_ansi_colors' by 'crayon::num_ansi_colors' when loading 'easypar'</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span></span>
<span><span class="co">#&gt; <span style="color: #5F8700;">&lt;cevodata&gt; dataset: Demo data</span></span></span>
<span><span class="co">#&gt; <span style="color: #87AF5F;">Genome: unknown</span></span></span>
<span><span class="co">#&gt; SNV assays: snvs (default)</span></span>
<span><span class="co">#&gt; CNV assays: cnvs (default)</span></span>
<span><span class="co">#&gt; 4 cases, 4 samples, 1 sample per case</span></span>
<span><span class="co">#&gt; 16000 mutations total, 4000 +/- 0 mutations per case</span></span>
<span><span class="co">#&gt; Active models: powerlaw_fixed_subclones</span></span></code></pre></div>
<p>A list of all fitted models can be obtained with:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_model_names.html">get_model_names</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "SFS"                      "powerlaw_fixed"          </span></span>
<span><span class="co">#&gt; [3] "powerlaw_fixed_subclones"</span></span></code></pre></div>
<p>and the last fitted model is the active one:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/active_models.html">active_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "powerlaw_fixed_subclones"</span></span></code></pre></div>
<p>Models can be viewed with the <code><a href="../reference/get_models.html">get_models()</a></code> function:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="va">cd</span>, which <span class="op">=</span> <span class="st">"powerlaw_fixed"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 11</span></span></span>
<span><span class="co">#&gt;   sample_id model  component  from    to length     A     b alpha rsquared best </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1  power… Neutral … 0.045 0.095 0.05    46.6 <span style="text-decoration: underline;">2</span>339.     2    1.00  TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 2  power… Neutral … 0.245 0.295 0.05    15.1  360.     2    0.989 TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 3  power… Neutral … 0.285 0.335 0.050<span style="text-decoration: underline;">0</span> 104.   609.     2    0.986 TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 4  power… Neutral … 0.245 0.295 0.050<span style="text-decoration: underline;">0</span>  78.8  747.     2    0.996 TRUE</span></span></code></pre></div>
<p>The active model is returned if the <code>which</code> argument is
left empty:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 13 × 16</span></span></span>
<span><span class="co">#&gt;    sample_id model    component   from     to  length     A     b alpha rsquared</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Sample 1  binomia… Clone     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Sample 1  powerla… Neutral …  0.045  0.095  0.05    46.6 <span style="text-decoration: underline;">2</span>339.     2    1.00 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Sample 2  binomia… Clone     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Sample 2  binomia… Subclone… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Sample 2  powerla… Neutral …  0.245  0.295  0.05    15.1  360.     2    0.989</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Sample 3  binomia… Clone     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Sample 3  binomia… Subclone… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Sample 3  binomia… Subclone… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Sample 3  powerla… Neutral …  0.285  0.335  0.050<span style="text-decoration: underline;">0</span> 104.   609.     2    0.986</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Sample 4  binomia… Clone     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Sample 4  binomia… Subclone… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Sample 4  binomia… Subclone… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> Sample 4  powerla… Neutral …  0.245  0.295  0.050<span style="text-decoration: underline;">0</span>  78.8  747.     2    0.996</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 6 more variables: best &lt;lgl&gt;, N &lt;int&gt;, cellularity &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   N_mutations &lt;int&gt;, BIC &lt;dbl&gt;, sequencing_DP &lt;dbl&gt;</span></span></span></code></pre></div>
<p>The tibble lists the components for all models and samples. Binomial
components are called either the clones or the subclones. The component
with the highest cellular frequency is called the clone, and the
remaining components are called subclones. This distinction is important
when estimating the subclonal evolutionary parameters: the main clones
are not under positive selection in the tumor, so we only estimate the
evolutionary parameters for the subclones.</p>
</div>
<div class="section level2">
<h2 id="plot-models">Plot models<a class="anchor" aria-label="anchor" href="#plot-models"></a>
</h2>
<p>The fitted models can be visualized with the
<code><a href="../reference/plot_models.html">plot_models()</a></code> function:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_models.html">plot_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>male <span class="op">=</span> <span class="st">"#DD4124"</span>, female <span class="op">=</span> <span class="st">"#00496F"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"cevomod models"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in geom_bar(join_aes(bar_mapping, mapping), stat = "identity", alpha =</span></span>
<span><span class="co">#&gt; alpha, : Ignoring unknown aesthetics: <span style="color: #00BB00;">width</span></span></span></code></pre></div>
<p><img src="get_started_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<p>The model fitted to sample 2 is very inaccurate. The slope of the
distribution is too steep to be approximated by the power-law exponent
of 2. This sample should not be fitted with the model with the fixed
power-law component. We will ignore this fit when estimating the
evolutionary parameters.</p>
</div>
<div class="section level2">
<h2 id="evolutionary-parameters">Evolutionary parameters<a class="anchor" aria-label="anchor" href="#evolutionary-parameters"></a>
</h2>
<p>The evolutionary parameters for the subclones can be calculated using
the equations provided by <a href="https://doi.org/10.1038/s41588-018-0128-6" class="external-link">Williams et
al. (2018)</a>. Williams assumes an exponential tumor growth and a
constant mutation rate, under which the power-law exponent equals 2, so
we can use the equations with the models we fitted. In cevomod, we use
the code and functions implemented in the <a href="https://github.com/caravagnalab/mobster/blob/master/R/evodynamics.R" class="external-link">MOBSTER
package</a> to calculate the parameters. We should not use the model for
Sample 2, though. The mutation rate is severely underestimated, and the
model fit is inaccurate. One should evaluate the fitted models
carefully, before continuing with the analysis.</p>
<p>Mutation rates can be obtained with
<code><a href="../reference/evo_params.html">get_mutation_rates()</a></code>, and the selection coefficients with
<code><a href="../reference/evo_params.html">get_selection_coefficients()</a></code> functions. Both functions can
be run on the cevodata objects (with the correct models fitted), or on
the model tibbles directly. This allows us to manually correct the model
tibbles before the calculation of the evolutionary parameters.</p>
<p>For example, one can get filter Sample 2 out from the cevodata
object:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample_id</span> <span class="op">!=</span> <span class="st">"Sample 2"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/evo_params.html">get_mutation_rates</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">#&gt;   sample_id mutation_rate_williams</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1                    46.6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 3                   104. </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 4                    78.8</span></span></code></pre></div>
<p>or from the models tibble:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample_id</span> <span class="op">!=</span> <span class="st">"Sample 2"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/evo_params.html">get_selection_coefficients</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 8</span></span></span>
<span><span class="co">#&gt;   sample_id mutation_rate_williams component  N_mutations subclone_frequency</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 3                   104.  Subclone 1         724              0.982</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 3                   104.  Subclone 2         785              0.236</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 4                    78.8 Subclone 1         824              1.00 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 4                    78.8 Subclone 2         366              0.183</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: emergence_time &lt;dbl&gt;, time_end &lt;dbl&gt;, selection &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Paweł Kuś.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
