<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cevomod">
<title>Get Started • cevomod</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Started">
<meta property="og:description" content="cevomod">
<meta property="og:image" content="https://pawelqs.github.io/cevomod/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">cevomod</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.9.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/get_started.html">Get Started</a>
    <a class="dropdown-item" href="../articles/cevodata_class.html">cevodata class</a>
    <a class="dropdown-item" href="../articles/fitting_models.html">Fitting models</a>
    <a class="dropdown-item" href="../articles/visualizations.html">Visualizations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get Started</h1>
            
      
      
      <div class="d-none name"><code>get_started.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="cevodata">cevodata<a class="anchor" aria-label="anchor" href="#cevodata"></a>
</h2>
<p><strong>Page under construction</strong></p>
<p>The main data structure used by cevomod is the cevodata S3 class. The
easiest way to prepare the object is to create an empty object using the
<code><a href="../reference/init_cevodata.html">init_cevodata()</a></code> constructor and populating the object with
the data using the <code>add_*_data()</code> methods.</p>
<p>Let’s use the list of SNV variants and the metadata from the small
attached artificial dataset <code>test_data</code></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pawelqs.github.io/cevomod/">cevomod</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">snvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assays.html">SNVs</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span></span>
<span><span class="va">sample_data</span> <span class="op">&lt;-</span> <span class="va">test_data</span><span class="op">$</span><span class="va">metadata</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snvs</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">&lt;cevo_snvs&gt; tibble</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 16,000 × 11</span></span></span>
<span><span class="co">#&gt;    sample_id chrom   pos gene_symbol ref   alt   ref_reads alt_reads   VAF</span></span>
<span><span class="co">#&gt;  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Sample 1  chr1      1 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           61         1 0.031</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Sample 1  chr1      2 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           37        29 0.441</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Sample 1  chr1      3 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           35        28 0.446</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Sample 1  chr1      4 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           54         3 0.068</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Sample 1  chr1      5 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           25        27 0.519</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Sample 1  chr1      6 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           24        31 0.562</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Sample 1  chr1      7 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           21         1 0.082</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Sample 1  chr1      8 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           22        24 0.521</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Sample 1  chr1      9 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           65         1 0.029</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Sample 1  chr1     10 <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>           25         6 0.211</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 15,990 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: impact &lt;chr&gt;, DP &lt;int&gt;</span></span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_data</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 3</span></span></span>
<span><span class="co">#&gt;   sample_id patient_id sex   </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1  Patient 1  male  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 2  Patient 2  male  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 3  Patient 3  female</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 4  Patient 4  male</span></span></code></pre></div>
<p>Most of the cevomod functions are pipe-oriented, so we can create and
populate the object through the small pipeline:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/init_cevodata.html">init_cevodata</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Demo data"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/assays.html">add_SNV_data</a></span><span class="op">(</span><span class="va">snvs</span>, name <span class="op">=</span> <span class="st">"snvs"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/cevo_metadata.html">add_sample_data</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cd</span></span>
<span><span class="co">#&gt; <span style="color: #5F8700;">&lt;cevodata&gt; dataset: Demo data</span></span></span>
<span><span class="co">#&gt; <span style="color: #87AF5F;">Genome: unknown</span></span></span>
<span><span class="co">#&gt; SNV assays: snvs (default)</span></span>
<span><span class="co">#&gt; CNV assays: None</span></span>
<span><span class="co">#&gt; 4 cases, 4 samples, 1 sample per case</span></span>
<span><span class="co">#&gt; 16000 mutations total, 4000 +/- 0 mutations per case</span></span>
<span><span class="co">#&gt; Active models:</span></span></code></pre></div>
<p><code>name</code> can be any string that is informative for the
user.</p>
</div>
<div class="section level2">
<h2 id="vaf-spectra">VAF spectra<a class="anchor" aria-label="anchor" href="#vaf-spectra"></a>
</h2>
<p>cevomod models are fitted to the Variant Allele Frequency Spectra of
the samples. Spectra can be displayed with <code><a href="../reference/sfs.html">plot_SFS()</a></code>
function (SFS - <em>Site Frequency Spectrum</em>). Most of the cevomod
plotting functions return a ggplot object which can be easily modified.
Additional aestetics can be added with the <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code> function.
Since the <code>metadata</code> is left-joined to the plot data, the
metadata columns can be directly used in the <code>eas()</code>
call.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sfs.html">plot_SFS</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>male <span class="op">=</span> <span class="st">"#DD4124"</span>, female <span class="op">=</span> <span class="st">"#00496F"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Variant Allele Frequency Spectrum"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in geom_bar(join_aes(bar_mapping, mapping), stat = "identity", alpha =</span></span>
<span><span class="co">#&gt; alpha, : Ignoring unknown aesthetics: <span style="color: #00BB00;">width</span></span></span></code></pre></div>
<p><img src="get_started_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="preparation">Preparation<a class="anchor" aria-label="anchor" href="#preparation"></a>
</h2>
<p>Before the the models are fitted, we need to run the
<code><a href="../reference/prepare_SNVs.html">prepare_SNVs()</a></code> function. In this step VAF values are
binarized, and the <code>VAF_interval</code> column is added. By
default, the number of bins equals the median sequencing coverage of
variants in the sample. It allows to reduce the aliasing noise in the
samples with low-sequencing depth, and allow to analyze the samples with
the deeper coverage with a greater resolution. The desired number of
bins can also be specified manually, using the <code>bins</code>
argument.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/prepare_SNVs.html">prepare_SNVs</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/assays.html">SNVs</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 16,000</span></span>
<span><span class="co">#&gt; Columns: 12</span></span>
<span><span class="co">#&gt; $ sample_id    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Sample 1", "Sample 1", "Sample 1", "Sample 1", "Sample 1…</span></span>
<span><span class="co">#&gt; $ chrom        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "chr1", "chr1", "chr1", "chr1", "chr1", "chr1", "chr1", "…</span></span>
<span><span class="co">#&gt; $ pos          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17…</span></span>
<span><span class="co">#&gt; $ gene_symbol  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…</span></span>
<span><span class="co">#&gt; $ ref          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…</span></span>
<span><span class="co">#&gt; $ alt          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…</span></span>
<span><span class="co">#&gt; $ ref_reads    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 61, 37, 35, 54, 25, 24, 21, 22, 65, 25, 31, 30, 58, 13, 2…</span></span>
<span><span class="co">#&gt; $ alt_reads    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 29, 28, 3, 27, 31, 1, 24, 1, 6, 18, 33, 1, 1, 39, 30, …</span></span>
<span><span class="co">#&gt; $ VAF          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.031, 0.441, 0.446, 0.068, 0.519, 0.562, 0.082, 0.521, 0…</span></span>
<span><span class="co">#&gt; $ impact       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "MODIFIER", "MODIFIER", "MODIFIER", "MODIFIER", "MODIFIER…</span></span>
<span><span class="co">#&gt; $ VAF_interval <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "(0.0196,0.0392]", "(0.431,0.451]", "(0.431,0.451]", "(0.…</span></span>
<span><span class="co">#&gt; $ DP           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 62, 66, 63, 57, 52, 55, 22, 46, 66, 31, 49, 63, 59, 14, 6…</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="model-fitting">Model fitting<a class="anchor" aria-label="anchor" href="#model-fitting"></a>
</h2>
<p>When the SNVs are prepared, the models can be fitted with the
<code>fit_*()</code> functions. Full model consist of the power-law
component and one or more binomial components, which are fitted
sequentially.</p>
<p>In this example, we first fit the power-law model with the exponent
equal to 2, and then the mixture of binomial models.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="va">cd</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/powerlaw_fixed_model.html">fit_powerlaw_tail_fixed</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/fit_subclones.html">fit_subclones</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Fitting binomial models</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Fitting williams neutral models...</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Mf_1f's not calculated yet. Calculating with default bins</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">Calculating Williams's M(f) ~ 1/f statistics</span></span></span>
<span><span class="co">#&gt; <span style="color: #5FAFD7;">SFS's not calculated yet. Calculating with sample DP as number of bins</span></span></span>
<span></span>
<span><span class="va">cd</span></span>
<span><span class="co">#&gt; <span style="color: #5F8700;">&lt;cevodata&gt; dataset: Demo data</span></span></span>
<span><span class="co">#&gt; <span style="color: #87AF5F;">Genome: unknown</span></span></span>
<span><span class="co">#&gt; SNV assays: snvs (default)</span></span>
<span><span class="co">#&gt; CNV assays: None</span></span>
<span><span class="co">#&gt; 4 cases, 4 samples, 1 sample per case</span></span>
<span><span class="co">#&gt; 16000 mutations total, 4000 +/- 0 mutations per case</span></span>
<span><span class="co">#&gt; Active models: powerlaw_fixed_subclones</span></span></code></pre></div>
<p>The fitted models can be listed with:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_model_names.html">get_model_names</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "powerlaw_fixed"           "powerlaw_fixed_subclones"</span></span></code></pre></div>
<p>and the last fitted model is the active one:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span><span class="op">$</span><span class="va">active_models</span></span>
<span><span class="co">#&gt; [1] "powerlaw_fixed_subclones"</span></span></code></pre></div>
<p>Models can be viewed with the <code><a href="../reference/get_models.html">get_models()</a></code> function:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="va">cd</span>, which <span class="op">=</span> <span class="st">"powerlaw_fixed"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 11</span></span></span>
<span><span class="co">#&gt;   sample_id model  component  from    to length     A     b alpha rsquared best </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1  willi… Neutral … 0.185 0.235   0.05  42.8 <span style="text-decoration: underline;">2</span>348.     2    0.980 TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 2  willi… Neutral … 0.215 0.265   0.05  25.5  299.     2    0.989 TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 3  willi… Neutral … 0.285 0.335   0.05 102.   595.     2    0.982 TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Sample 4  willi… Neutral … 0.265 0.315   0.05  67.4  800.     2    0.980 TRUE</span></span></code></pre></div>
<p>The active model is returned if the <code>which</code> argument is
left empty:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 11 × 18</span></span></span>
<span><span class="co">#&gt;    sample_id model     component   from     to length     A     b alpha rsquared</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Sample 1  binomial… Clone     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Sample 1  binomial… Subclone… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Sample 1  williams  Neutral …  0.185  0.235   0.05  42.8 <span style="text-decoration: underline;">2</span>348.     2    0.980</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Sample 2  binomial… Clone     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Sample 2  williams  Neutral …  0.215  0.265   0.05  25.5  299.     2    0.989</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Sample 3  binomial… Clone     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Sample 3  binomial… Subclone… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Sample 3  williams  Neutral …  0.285  0.335   0.05 102.   595.     2    0.982</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Sample 4  binomial… Clone     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Sample 4  binomial… Subclone… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Sample 4  williams  Neutral …  0.265  0.315   0.05  67.4  800.     2    0.980</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 8 more variables: best &lt;lgl&gt;, N &lt;int&gt;, cellularity &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   N_mutations &lt;dbl&gt;, BIC &lt;dbl&gt;, sequencing_DP &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   has_overlapping_clones &lt;lgl&gt;, max_BIC &lt;dbl&gt;</span></span></span></code></pre></div>
<p>The tibble lists all parameters of all modeled samples. Bino</p>
</div>
<div class="section level2">
<h2 id="plot-models">Plot models<a class="anchor" aria-label="anchor" href="#plot-models"></a>
</h2>
<p>The fitted models can be visualized with the
<code><a href="../reference/plot_models.html">plot_models()</a></code> function:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_models.html">plot_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>male <span class="op">=</span> <span class="st">"#DD4124"</span>, female <span class="op">=</span> <span class="st">"#00496F"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Variant Allele Frequency Spectrum"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in geom_bar(join_aes(bar_mapping, mapping), stat = "identity", alpha =</span></span>
<span><span class="co">#&gt; alpha, : Ignoring unknown aesthetics: <span style="color: #00BB00;">width</span></span></span></code></pre></div>
<p><img src="get_started_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>Model fitted to the Sample 2 is very inacurrate. The slope of the
distribution is too step to be approximated with the power-law exponent
of 2. This sample should not be fitted the model with the fixed
power-law component. For the rest of samples we can use the equations
provided by <a href="https://www.nature.com/articles/s41588-018-0128-6" class="external-link">Williams et
al.</a> to estimate the evolutionary parameters of the subclones.</p>
</div>
<div class="section level2">
<h2 id="evolutionary-parameters">Evolutionary parameters<a class="anchor" aria-label="anchor" href="#evolutionary-parameters"></a>
</h2>
<p>The evolutionary parameters are estimated for the subclonal
components of the models. The binomial components with the highest
frequency are called Clones and no parameters are estimated for
them.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_models.html">get_models</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">component</span>, <span class="st">"Subclone"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 18</span></span></span>
<span><span class="co">#&gt;   sample_id model  component  from    to length     A     b alpha rsquared best </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1  binom… Subclone…    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span> TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 3  binom… Subclone…    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span> TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 4  binom… Subclone…    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>       <span style="color: #BB0000;">NA</span> TRUE </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7 more variables: N &lt;int&gt;, cellularity &lt;dbl&gt;, N_mutations &lt;dbl&gt;, BIC &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   sequencing_DP &lt;dbl&gt;, has_overlapping_clones &lt;lgl&gt;, max_BIC &lt;dbl&gt;</span></span></span></code></pre></div>
<p>For the model with the power-law exponent equal to 2, have proposed
an equations to estimate the evolutionary parameters such as</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/evo_params.html">get_selection_coefficients</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 8</span></span></span>
<span><span class="co">#&gt;   sample_id mutation_rate_williams component  N_mutations subclone_frequency</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Sample 1                    42.8 Subclone 1          62              0.233</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Sample 3                   102.  Subclone 1         826              0.275</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Sample 4                    67.4 Subclone 1         546              0.222</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: emergence_time &lt;dbl&gt;, time_end &lt;dbl&gt;, selection &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Paweł Kuś.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
